mediapipe/tasks/ios/genai/core/sources/LlmSessionRunner.swift

```
@@ -0,0 +1,209 @@
+  /// - Throws: An error if the LLM's response is invalid.
+  func predict() throws -> [String] {
+    /// No safe guards for the call since the C++ APIs only throw fatal errors.
+    /// `LlmInferenceEngine_Session_PredictSync()` will always return a `LlmResponseContext` if the
+    /// call completes.
+    var responseContext = LlmInferenceEngine_Session_PredictSync(cLlmSession)
+
+    defer {
+      withUnsafeMutablePointer(to: &responseContext) {
+        LlmInferenceEngine_CloseResponseContext($0)
+      }
+    }
+
+    /// Throw an error if response is invalid.
+    guard let responseStrings = LlmSessionRunner.responseStrings(from: responseContext) else {
+      throw GenAiInferenceError.invalidResponse
+    }
+
+    return responseStrings
+  }
+
+  /// Invokes the C LLM session with the previously added query chunks asynchronously to generate an
+  /// array of `String` responses from the LLM. The `progress` callback returns the partial 
+  /// responses from the LLM or any errors. `completion` callback is invoked once the LLM is done 
+  /// generating responses.
+  ///
+  /// - Parameters:
+  ///   - progress: A callback invoked when a partial response is available from the C LLM Session.
+  ///   - completion: A callback invoked when the C LLM Session finishes response generation.
+  /// - Throws: An error if the LLM's response is invalid.
+  func predictAsync(
+    progress: @escaping (_ partialResult: [String]?, _ error: Error?) -> Void,
+    completion: @escaping (() -> Void)
+  ) throws {
+    let callbackInfo = CallbackInfo(progress: progress, completion: completion)
+    let callbackContext = UnsafeMutableRawPointer(Unmanaged.passRetained(callbackInfo).toOpaque())
+
+    LlmInferenceEngine_Session_PredictAsync(cLlmSession, callbackContext) {
+      context, responseContext in
+      guard let cContext = context else {
+        return
+      }
+      guard let cResponse = responseContext?.pointee else {
+        return
+      }
```

Is this expected? Shall we throw an error here?